lib/rucio/transfertool/transfertool.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2018 CERN for the benefit of the ATLAS collaboration.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+# - Martin Barisits <martin.barisits@cern.ch>, 2018
+
+from abc import ABCMeta, abstractmethod
+
+
+class Transfertool():
+    """
+    Interface definition of the Rucio transfertool
+    """
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, external_host):
+        """
+        Initializes the transfertool
+
+        :param external_host:   The external host where the transfertool API is running
+        """
+
+        self.external_host = external_host
+
+    @abstractmethod
+    def submit(self, files, job_params, timeout=None):
+        """
+        Submit transfers to the transfertool.
+
+        :param files:        List of dictionaries describing the file transfers.
+        :param job_params:   Dictionary containing key/value pairs, for all transfers.
+        :returns:            Transfertool internal identifiers.
+        """
+        pass
+
+    @abstractmethod
+    def query(self, transfer_ids, details=False):
```

Comment:
@param timeout in all methods is a good idea

Reply:
About`transfer_ids`, if the transfer tool is simple webdav 3rd party transfer command or local cp, then `tranfersIds` are actually the same than `requestids`, correct ? btw Where is done the mapping destination=transferstool implementation ?

Reply:
Will add.

Reply:
the generation of the transfer_ids is really up to the transfertool. But in theory a simple transfertool could just hand back the request_ids yes.
The mapping must be done in the conveyor, but from these code changes we are a little bit away still.